src/game/scenes/arena.c

```diff
@@ -1539,7 +1540,9 @@ int arena_create(scene *scene) {
             local->rec->pilots[i].info.color_1 = player->pilot->color_1;
             local->rec->pilots[i].info.color_2 = player->pilot->color_2;
             local->rec->pilots[i].info.color_3 = player->pilot->color_3;
-            memcpy(local->rec->pilots[i].info.name, lang_get(player->pilot->pilot_id + 20), 18);
+            // XXX ugly strncpy, implicit truncation, no nul termination guarantee, why are we even copying the pilot's
+            // name in here?
+            strncpy(local->rec->pilots[i].info.name, lang_get_offset(LangPilot, player->pilot->pilot_id), 18);
```

Comment:
There is that `strncpy_or_truncate` that could used now.

Reply:
Added and using strncpy_or_assert, which asserts on truncation.